torch/_inductor/codegen/cpp_gemm_template.py

```diff
@@ -169,13 +182,19 @@ def __init__(
         self.alpha = alpha
         self.has_bias = has_bias
         self.register_blocking = register_blocking
-        m, n = layout.size
-        _, k = input_nodes[0].get_size()
+        m, n = layout.size[-2:]
+        k = input_nodes[0].get_size()[-1]
         self.m, self.n, self.k = m, n, k
         self.is_dynamic_M = has_free_symbols((m,))
-
-    @cache_on_self
-    def thread_blocking(self) -> GemmBlocking:
+        self.should_pack_weights = True
+
+    def thread_blocking(self, num_threads=None, reset_cache=False) -> GemmBlocking:
+        if hasattr(self, "_thread_blocking_cache") and not reset_cache:
+            thread_count, cache = self._thread_blocking_cache
+            if thread_count == num_threads:
+                return cache
```

Comment:
To make things consistent, please refer to `torch/_inductor/sizevars.py` how to implement the caching of a member function with args, e.g., `make_stride_vars_cache`.